### PR TITLE
[Feature] Side Nav External Links

### PR DESCRIPTION
**Pull request by patryk0605**
This PR introduces the ability to support external links in the side nav.

Closes: https://github.com/mobi/goponents/issues/211

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.scss [hunk 1]
```diff
@@ -5,7 +5,6 @@
 .go-nav-item {
   align-items: center;
   display: flex;
-  padding-right: .5rem;
```
Comment (patryk0605) [inline projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.scss]: Wasn't sure of the purpose of this and was making it impossible to align `launch` icon with dropdown arrow

File: projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.scss [hunk 2]
```diff
@@ -58,6 +57,7 @@
   font-weight: $weight-light;
   letter-spacing: $side-nav-letter-spacing;
   padding: $inner-side-nav-padding;
+  flex-grow: 1;
```
Comment (patryk0605) [inline projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.scss]: Added to push icons to the right.

File: projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.html [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
-             *ngIf="navItem.routeIcon">
+    <ng-container *ngTemplateOutlet="navItemInner; context: { $implicit: navItem }"></ng-container>
+  </a>
+
+  <a class="go-nav-item__link"
+     *ngIf="navItem.isExternalLink"
+     [href]="navItem.route"
+     [target]="navItem.externalLinkTarget ? navItem.externalLinkTarget : '_blank'"
+     rel="noreferrer">
+    <ng-container *ngTemplateOutlet="navItemInner; context: { $implicit: navItem }"></ng-container>
```
[the remark this answers is not in the document]
Comment (patryk0605) [reply projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.html]: We don't need context here, I just put it out of habit. Makes sense for larger complicated components, but you're right that here we don't need it as not much is going on. I personally like to use context so it's easier to understand what data `ng-template` is using, but it's personal preference 😄

File: projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.html [hunk 4]
```diff
@@ -1,16 +1,34 @@
...
-    <go-icon [icon]="navItem.routeIcon"
-             iconClass="go-nav-group__icon"
-             *ngIf="navItem.routeIcon">
+    <ng-container *ngTemplateOutlet="navItemInner; context: { $implicit: navItem }"></ng-container>
+  </a>
+
+  <a class="go-nav-item__link"
+     *ngIf="navItem.isExternalLink"
+     [href]="navItem.route"
+     [target]="navItem.externalLinkTarget ? navItem.externalLinkTarget : '_blank'"
```
[the remark this answers is not in the document]
Comment (patryk0605) [reply projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.html]: If we default it in the component, wouldn't it be the same as what I have going on now anyway? `externalLinkTarget` is being passed in with the `NavItem` object. So within the component I would have to do: 

```
@Input() navItem: NavItem;
externalLinkTarget: '_self' | '_blank' | '_parent' | '_top' = navItem.externalLinkTarget ? navItem.externalLinkTarget : '_blank';
```

Not unless I'm misunderstanding what you're asking for 😃 as I see in your example, you're setting the default within what seems to be the interface, but Typescript does not allow interfaces to have an initializer (or do they now 😋).

**Comment by patryk0605** [comment]
@StevenUlmer PTAL - I addressed one of your comments and had a question about the other.

File: projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (StevenUlmer) [reply projects/go-lib/src/lib/components/go-side-nav/go-nav-item/go-nav-item.component.html]: Nope you're correct. I missed that this code was in the interface and not a binding on the component. This is fine. We could also do this in in the onit by doing `navItem.externalLinkTarget = navItem.externalLinkTarget || 'blank';`, but this is fine for now

**Pull request merged**